src/rules/unnecessaryElseRule.ts

```diff
@@ -46,20 +57,28 @@ export class Rule extends Lint.Rules.AbstractRule {
     }
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithFunction(sourceFile, walk);
+        return this.applyWithFunction(sourceFile, walk, {
+            allowElseIf: this.ruleArguments.indexOf(OPTION_ALLOW_ELSE_IF) !== -1,
+        });
     }
 }
 
-function walk(ctx: Lint.WalkContext<void>): void {
+interface Options {
+    allowElseIf: boolean;
+}
+
+function walk(ctx: Lint.WalkContext<Options>): void {
     ts.forEachChild(ctx.sourceFile, function cb(node: ts.Node): void {
         if (utils.isIfStatement(node)) {
             const jumpStatement = utils.isBlock(node.thenStatement)
                 ? getJumpStatement(getLastStatement(node.thenStatement))
                 : getJumpStatement(node.thenStatement);
 
             if (jumpStatement !== undefined && node.elseStatement !== undefined) {
-                const elseKeyword = getPositionOfElseKeyword(node, ts.SyntaxKind.ElseKeyword);
-                ctx.addFailureAtNode(elseKeyword, Rule.FAILURE_STRING(jumpStatement));
+                if (!utils.isIfStatement(node.elseStatement) || !ctx.options.allowElseIf) {
```

Comment:
Nit: you can combine the two `if` statements into one for a little more conciseness.

Reply:
@JoshuaKGoldberg just for my knowledge: what does "nit" mean? 🙂

Reply:
Ah, it stands for "nitpick" 😀